File: data/EXP01/golang_no_system_var/2358/2358.output_1.go
func maximumGroups(grades []int) int {
    sort.Ints(grades)
    
    sum := grades[0]
    count := 1
    groupCount := 1
    
    for i := 1; i < len(grades); i++ {
        if sum > grades[i] || count > i {
            groupCount++
            sum = grades[i]
            count = 1
        } else {
            sum += grades[i]
            count++
        }
    }
    
    return groupCount
}

File: data/EXP01/python/2358/2358.output_1.py
# @lc app=leetcode id=2358 lang=python3
class Solution:
    def maximumGroups(self, grades: List[int]) -> int:
        grades.sort(reverse=True)  # Sort the array in descending order

        groups = 1  # Number of groups formed
        prev_sum = grades[0]  # Sum of grades in the previous group

        for i in range(1, len(grades)):
            if grades[i] * (i+1) > prev_sum:
                groups += 1
                prev_sum += grades[i]
                
        return groups